icu4c/source/common/cmemory.h

```diff
@@ -728,6 +735,28 @@ class MemoryPool : public UMemory {
         return fPool[fCount++] = new T(std::forward<Args>(args)...);
     }
 
+    /**
+     * Takes ownership of ptr, adding it to the pool.
+     *
+     * @param ptr Pointer to take ownership of.
+     * @param status Is set to U_MEMORY_ALLOCATION_ERROR if allocation of the
+     * needed pool capacity fails.
+     */
+    template<typename... Args>
+    void adoptBack(T* ptr, UErrorCode& status) {
```

Comment:
I would make the MemoryPool version "adopt", since MemoryPool has no concept of order.